lib/seattleflu/id3c/cli/command/clinical.py

```diff
@@ -874,35 +874,84 @@ def parse_kp2023(kp2023_filename: str) -> None:
+        missing_cols.remove('symptom__throat')
+    if len(missing_cols) > 0:
+        raise MissingColumn(f'One or more expected columns are missing from the input spreadsheet: {*missing_cols,}')
 
     # rename columns
     column_map = {
-        'marshfield_lab_id':    'collection_id', # will be mapped to lims barcode during id3c clinical match-kp2023
+        'marshfield_lab_id':    'collection_id', # will be mapped to lims barcode with id3c clinical match-kp2023
         'hispaniclatino':       'ethnicity',
-        'assigned_sex':          'sex',
-        'symptom__throat':      'symptom_throat', # fix extra underscore in column name
+        'assigned_sex':         'sex',
+        'symptom__throat':      'symptom_throat', # fix extra underscore
         'censustract':          'census_tract',
         'type_of_visit':        'patient_class',
-        'sympton_sob':          'symptom_sob'
+        'sympton_sob':          'symptom_sob' # fix typo
     }
 
     clinical_records = clinical_records.rename(columns=column_map)
 
     # check for missing or duplicated barcodes?
     #barcode_quality_control(clinical_records)
 
-    # in the lims, the marshfield_lab_id / collection_id has an aliquot number appended to the end of the id.
-    # for example, KPWC100000-1
-    # as of Oct 2023, the collection ids from Kaiser in the metadata excel files do not have this aliquot number appended to them
-    # so we want to check if the aliquot id is present, and if not, append it
-    clinical_records['collection_id'] = clinical_records['collection_id'].apply(lambda x: x if re.search(r'-\d+$', x) else x+'-1')
+    # The collection ids on the tubes from KP have aliquot numbers appended to them (ex: KPWB100001C-1)
+    # but the collection ids in the metadata spreadsheet do not have these aliquot numbers at the end (ex: KPWB100001C)
+    # therefore, we will check that there is no aliquot number at the end of the collection id in the metadata spreadsheet,
+    # and strip the aliquot number if it is there, with a warning since we don't expect it
+    # in the LIMS, this id is called KaiserPermanenteSpecimenId and will also have its aliquot number stripped
+    collection_ids_with_aliquot = clinical_records.loc[clinical_records['collection_id'].apply(lambda x: True if re.search(r'-\d+$', x) else False)]['collection_id'].values
+    # if there are any collection ids with aliquot (not expected), give a warning and strip the aliquot number
+    if len(collection_ids_with_aliquot) > 0:
+        LOG.warning(f'Warning: One or more Marshfield lab IDs contain aliquot id, which is unexpected. Stripping aliquot id from these samples: {*collection_ids_with_aliquot,}')
+        for cid in collection_ids_with_aliquot:
+            # strip aliquot number ('-<number>') from collection id
+            clinical_records.loc[clinical_records['collection_id'] == cid, 'collection_id'] = re.sub(r'-\d+$','',cid)
```

Comment:
You may find this less readable, so feel free to ignore, but when I am working with Pandas dataframes I will generally try to avoid for loops. You can do a lot of the same things with internal Pandas functions and indexing, which have some pretty significant performance benefits since it is all C compiled code. Obviously here those performance gains don't matter so much, but for transparency here is a way you can keep this logic within Pandas.

 ```suggestion
    collection_ids_with_aliquot = clinical_records['collection_id'].apply(lambda x: True if re.search(r'-\d+$', x) else False)
    # if there are any collection ids with aliquot (not expected), give a warning and strip the aliquot number
    if any(collection_ids_with_aliquot):
        LOG.warning(
            'Warning: One or more Marshfield lab IDs contain aliquot id, which is unexpected. Stripping aliquot id from' +
            f'these samples: {*clinical_records.loc[collection_ids_with_aliquot]["collection_id"].values,}'
        )
        clinical_records.loc[collection_ids_with_aliquot, 'collection_id'] = clinical_records.loc[
            collection_ids_with_aliquot, 'collection_id'
        ].apply(lambda cid: re.sub(r'-\d+$','', cid))
```